recipe/meta.yaml

```diff
@@ -81,128 +75,155 @@ requirements:
-        {{ opencv_libs.append("stereo") or "" }}
-        {{ opencv_libs.append("stitching") or "" }}
-        {{ opencv_libs.append("structured_light") or "" }}
-        {{ opencv_libs.append("superres") or "" }}
-        {{ opencv_libs.append("surface_matching") or "" }}
-        {{ opencv_libs.append("text") or "" }}
-        {{ opencv_libs.append("tracking") or "" }}
-        {{ opencv_libs.append("video") or "" }}
-        {{ opencv_libs.append("videoio") or "" }}
-        {{ opencv_libs.append("videostab") or "" }}
-        {{ opencv_libs.append("wechat_qrcode") or "" }}
-        {{ opencv_libs.append("xfeatures2d") or "" }}
-        {{ opencv_libs.append("ximgproc") or "" }}
-        {{ opencv_libs.append("xobjdetect") or "" }}
-        {{ opencv_libs.append("xphoto") or "" }}
-        {{ opencv_libs.append("freetype") or "" }}
-        - export MACOSX_DEPLOYMENT_TARGET={{ MACOSX_DEPLOYMENT_TARGET }}       # [osx]
-        - export CONDA_BUILD_SYSROOT={{ CONDA_BUILD_SYSROOT }}                 # [osx]
-        - OPENCV_FLAGS=`pkg-config --cflags opencv4`  # [unix]
-        - $CXX -std=c++11 $RECIPE_DIR/test.cpp ${OPENCV_FLAGS} -o test   # [unix]
-        - if [[ $(./test) != $PKG_VERSION ]]; then exit 1 ; fi                # [unix]
-        {% for each_opencv_lib in opencv_libs %}
-        - echo Testing for presence of {{ each_opencv_lib }}
-        - test -f $PREFIX/lib/libopencv_{{ each_opencv_lib }}${SHLIB_EXT}                  # [unix]
-        - if not exist %PREFIX%\\Library\\bin\\opencv_{{ each_opencv_lib }}{{ win_ver_lib }}.dll exit 1  # [win]
+        {% set opencv_libs = [
+            "alphamat",
+            "aruco",
+            "bgsegm",
+            "calib3d",
+            "ccalib",
+            "core",
+            "datasets",
+            "dnn",
+            "dnn_objdetect",
+            "dnn_superres",
+            "dpm",
+            "face",
+            "features2d",
+            "flann",
+            "freetype",
+            "fuzzy",
+            "gapi",
+            "hfs",
+            "highgui",
+            "img_hash",
+            "imgcodecs",
+            "imgproc",
+            "intensity_transform",
+            "line_descriptor",
+            "mcc",
+            "ml",
+            "objdetect",
+            "optflow",
+            "phase_unwrapping",
+            "photo",
+            "plot",
+            "quality",
+            "rapid",
+            "reg",
+            "rgbd",
+            "saliency",
+            "shape",
+            "stereo",
+            "stitching",
+            "structured_light",
+            "superres",
+            "surface_matching",
+            "text",
+            "tracking",
+            "video",
+            "videoio",
+            "videostab",
+            "wechat_qrcode",
+            "xfeatures2d",
+            "ximgproc",
+            "xobjdetect",
+            "xphoto"
+        ] + (not win) * [
+            "bioinspired",
+            "hdf"
+        ]%}
```

Comment:
the old style i remember was used because the bot was having trouble with certain aspects of templating.

I think this recipe is always problematic for the auto-tick bot, but I just watned to bring this up as a potential reason why we might want to revert this particular change.

Reply:
I've had no problems with multiline jinja variables in places like [arrow](https://github.com/conda-forge/arrow-cpp-feedstock/blob/main/recipe/meta.yaml#L158-L171), and the auto-tickbot still manages to open new PRs: https://github.com/conda-forge/arrow-cpp-feedstock/pull/1087

I don't doubt that the problems were there, but 38d050881c3be2f91f3550437be320d81109938c is 2.5 years old. Perhaps things changed in the meantime?

It's not a crucial aspect of this PR, but I find the current wall of text to be noisy bordering on unreadable, so I'd like to try getting rid of it.

Reply:
just giving a historical perspective when i can. Excited to see references to PYTHON2 go away!